### PR TITLE
Fix Dashboard Reload issue

### PR DESCRIPTION
**Pull request by RedLeader16**
# Description

1. Primarily a docker/nginx issue. Also patches static files not being served correctly and removes prometheus queries from non-prod environments. 

This change helps alleviate the tedium of constantly navigating back to a resource when refreshing the browser to view new data/state. 

(tested in docker mode)
https://user-images.githubusercontent.com/34227093/233500405-d5fadb8f-654c-4816-ad92-93c183f04520.mov

2. Added further support for more developer options. 
You can override the dashboard's api url address by creating a `.env.local` file with the following variable.
.env.local -> NEXT_PUBLIC_REACT_APP_API_URL=http://127.0.0.1
For more, visit: [next.js env vars](https://nextjs.org/docs/basic-features/environment-variables).

In this way, you can run a local dashboard instance (either in docker or via node) and point to a separate running feature form instance (treating it like an external data source).  This speeds up the validation loop.
Video Example:
`localhost` - running featureform instance in docker container
`localhost:3007` - running dashboard instance  in docker container
`localhost:3000` - running dashboard instance via node (npm run dev)

https://user-images.githubusercontent.com/34227093/233746538-6fde916d-4d1b-49a0-a163-3e53bce0622a.mov

Unit Tests:

Screenshot 2023-04-24 at 9 32 12 AM

### Does this correspond to an open issue?

https://github.com/featureform/featureform/issues/748

### Select type(s) of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have fixed any merge conflicts

### CODE REVIEW

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
           url: "https://www.featureform.com/"
           # optional:
           honorRobotExclusions: false
-          ignorePatterns: "*linkedin*,*autodesk*,*crunchbase*" # Linkedin returns 999 error to web scrapers
+          ignorePatterns: "*linkedin*,*autodesk*,*crunchbase*,*springboard*" # Linkedin returns 999 error to web scrapers
```
Comment (RedLeader16) [inline .github/workflows/link-check.yml:42]: @MMBazel thoughts on adding springboard to the ignore list?

File: dashboard/src/api/resources/Resources.js [hunk 2]
```diff
@@ -152,8 +152,17 @@ const SEARCH_URL = {
   apiKey: "",
 };
 
-if (typeof process.env.REACT_APP_API_URL != "undefined") {
+console.log("Node ENV:", process.env.NODE_ENV);
```
Comment (epps) [inline dashboard/src/api/resources/Resources.js:155]: Do we want to leave these console logs in?

**Review by sdreyer — APPROVED** [review]
One question, otherwise lgtm

File: dashboard/src/api/resources/Resources.js (hunk 2 above)
Comment (RedLeader16) [reply dashboard/src/api/resources/Resources.js:155]: good catch, will remove!

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (RedLeader16) [reply .github/workflows/link-check.yml:42]: actually, since the job is flaky (and as you pointed out the link works), I'll revert this change.

**Review by sdreyer — APPROVED** [review]

**Pull request merged**